src/defs.hpp

```diff
@@ -105,7 +105,7 @@ struct RegionSize { // aggregate and POD type; do NOT reorder member declaration
 // X1DIR x, r, etc...
 // X2DIR y, theta, etc...
 // X3DIR z, phi, etc...
-enum CoordinateDirection { X0DIR = 0, X1DIR = 1, X2DIR = 2, X3DIR = 3 };
+enum CoordinateDirection { NODIR = -1, X0DIR = 0, X1DIR = 1, X2DIR = 2, X3DIR = 3 };
 
```

Comment:
`X1DIR` and friends are used for indexing into arrays in various places. Making them a class, and thus type-safe, would break that behaviour and/or make it much more verbose.